src/functions.js

```diff
@@ -357,6 +358,61 @@ const queryRegionHomes = async ({ qs, type, cat = 'cat1' }) => {
     };
 };
 
+/**
+ *
+ * @param {Apify.Request} request
+ * @param {keyof TYPES} inputType
+ * @param {Puppeteer.Page} page
+ * @param {any} pageQs
+ * @param {Number} paginationPage
+ * @returns query states with total count
+ */
+const extractQueryStates = async (request, inputType, page, pageQs, paginationPage = 1) => {
+    /** @type { { states: Array<any>, totalCount: Number } } */
+    const queryStates = {
+        states: [],
+        totalCount: 0,
+    };
+
+    const type = request.userData.searchQueryState ? 'qs' : inputType;
+    const qs = translateQsToFilter(request.userData.searchQueryState || pageQs.queryState);
+    qs.pagination = { currentPage: paginationPage };
+
+    const filterStates = getQueryFilterStates(qs, type);
+
+    const listingTypes = ['cat1', 'cat2']; // cat1 = agents listings, cat2 = other listings
+    for (const cat of listingTypes) {
+        const wants = {
+            [cat]: ['listResults', 'mapResults'],
+        };
+
+        for (const filterState of filterStates) {
+            qs.filterState = filterState;
+            const url = `https://www.zillow.com/search/GetSearchPageState.htm?searchQueryState=${JSON.stringify(qs)}&wants=${JSON.stringify(wants)}&requestId=${Math.floor(Math.random() * 70) + 1}`;
+            log.debug(`Fetching url: ${url}`);
+
+            const result = await page.evaluate(
+                queryRegionHomes,
+                {
+                    qs: translateQsToFilter(request.userData.searchQueryState || pageQs.queryState),
+                    cat,
+                },
+            );
+
+            log.debug('query', result.qs);
+
+            const searchState = JSON.parse(result.body);
+            queryStates.states.push({ qs, searchState });
+
+            await Apify.setValue('SEARCH_STATE', searchState);
```

Comment:
Here you are setting value in a loop overriding itself so that is useless. If it is just for debug, delete it or wrap it inside `if (debug) {}`